pkg/webhook/resources/virtualmachine/mutator.go

```diff
@@ -217,3 +247,196 @@ func (m *vmMutator) getOvercommit() (*settings.Overcommit, error) {
 	}
 	return overcommit, nil
 }
+
+func needUpdateAffinity(oldVm, newVm *kubevirtv1.VirtualMachine) bool {
+	if oldVm.Spec.Template == nil || newVm.Spec.Template == nil {
+		return false
+	}
+
+	if oldVm.Spec.Template.Spec.Affinity.String() != newVm.Spec.Template.Spec.Affinity.String() {
+		return true
+	}
+
+	oldVmNetworks, newVmNetworks := oldVm.Spec.Template.Spec.Networks, newVm.Spec.Template.Spec.Networks
+	if len(getMultusNetworkIncrement(oldVmNetworks, newVmNetworks)) != 0 ||
+		len(getMultusNetworkIncrement(newVmNetworks, oldVmNetworks)) != 0 {
+		return true
+	}
```

Comment:
Those condition checks will bypass the VMs created in the previous version, and their affinity will never be updated until they add those affinity rules manually.
We can assume that when a VM has a NIC(except the cluster network),  the affinity must not be nil.